classes/query/Repository.php

```diff
@@ -0,0 +1,218 @@
+     */
+    public function addQuery(int $submissionId, int $stageId, string $title, string $content, User $fromUser, array $participantUserIds, int $contextId, bool $sendEmail = true): int
+    {
+        $query = Query::create([
+            'assocType' => Application::ASSOC_TYPE_SUBMISSION,
+            'assocId' => $submissionId,
+            'stageId' => $stageId,
+            'seq' => REALLY_BIG_NUMBER
+        ]);
+
+        $this->resequence(Application::ASSOC_TYPE_SUBMISSION, $submissionId);
+
+        foreach ($participantUserIds as $participantUserId) {
+            QueryParticipant::create([
+                'queryId' => $query->id,
+                'userId' => $participantUserId
+            ]);
+        }
+
+        Note::create([
+            'assocType' => Application::ASSOC_TYPE_QUERY,
+            'assocId' => $query->id,
+            'contents' =>  $content,
+            'title' =>  $title,
+            'userId' =>  $fromUser->getId(),
+        ]);
+
+        // Add task for assigned participants
+        $notificationMgr = new NotificationManager();
+
+        /** @var NotificationSubscriptionSettingsDAO */
+        $notificationSubscriptionSettingsDao = DAORegistry::getDAO('NotificationSubscriptionSettingsDAO');
+
+        foreach ($participantUserIds as $participantUserId) {
+            $notificationMgr->createNotification(
+                Application::get()->getRequest(),
+                $participantUserId,
+                Notification::NOTIFICATION_TYPE_NEW_QUERY,
+                $contextId,
+                Application::ASSOC_TYPE_QUERY,
+                $query->id,
+                Notification::NOTIFICATION_LEVEL_TASK
+            );
+
+            if (!$sendEmail) {
+                continue;
+            }
+
+            // Check if the user is unsubscribed
+            $notificationSubscriptionSettings = $notificationSubscriptionSettingsDao->getNotificationSubscriptionSettings(
+                NotificationSubscriptionSettingsDAO::BLOCKED_EMAIL_NOTIFICATION_KEY,
+                $participantUserId,
+                $contextId
+            );
+            if (in_array(Notification::NOTIFICATION_TYPE_NEW_QUERY, $notificationSubscriptionSettings)) {
+                continue;
+            }
+
+            $recipient = Repo::user()->get($participantUserId);
+            $mailable = new Mailable();
+            $mailable->to($recipient->getEmail(), $recipient->getFullName());
+            $mailable->from($fromUser->getEmail(), $fromUser->getFullName());
+            $mailable->subject($title);
+            $mailable->body($content);
+
+            Mail::send($mailable);
+        }
+
+        return $query->id;
+    }
+
+    /**
+     * Create a query with a submission's comments for the editors
+     *
+     * Creates the query and assigns all participants
+     *
+     * @return int new query id
+     */
+    public function addCommentsForEditorsQuery(Submission $submission): int
```

Comment:
In progress - only current usage is in `lib/pkp/classes/submission/Repository.php`.